File: UCF HSPT Documents/2008/Solutions/pizza.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

int main()
{
    //file pointer that will open the input file
    FILE *ifp;
    
    //opens the input file
    ifp = fopen("pizza.in", "r");
    
    //will hold the number of test cases that will be executed
    int cases = 0;
    
    //reads in how test cases there are
    fscanf(ifp, "%d", &cases);
    
    int i = 0;
    
    //for loop to run through every test case in the input file
    for(i = 1; i <= cases; i++)
    {
          int p = 0;
          int t = 0;
          int c = 0;
          
          //reads in a single test case
          fscanf(ifp, "%d", &p);
          fscanf(ifp, "%d", &t);
          fscanf(ifp, "%d", &c);
          
          //using Circumference = 2 * pi * radius and Area = pi * radius ^ 2 
          //the equation becomes Circumference ^ 2 / (4 * pi) this is then
          //divided by the number of slices and then multiplied by the number
          //of slices that Perry ate this is multiplied by 10 for the calories
          //and then use the ceil function in the math header file to round 
          //the final answer up and finally cast the number to an integer
          int ans = ((int)ceil((((p * p)/(4 * 3.14159265))/t) * c * 10));
          
          //prints out the output for the file
          printf("Perfectly Popular Pizza %d: Perry consumed %d calories.\n", i, ans);
    }
}

File: UCF HSPT Team/src/y2008/pizza.java
import java.io.*;
import java.util.*;

public class pizza{

	static double pi = 3.14159265;

	public static void main(String[] args) throws Exception{

		// Sets up the file input and reads in the number of orders Perry ordered
		Scanner In = new Scanner(new File("pizza.in"));
		int numOrders = In.nextInt();
	
		for(int currOrder = 1;currOrder <= numOrders;currOrder++){

			/* Reads in p, t, c. The perimeter p is equal to 2*r*pi, where r is the radius of the pie.
			 * The area of a circular pie, a, is equal to r^2*pi. So if you rewrite the equation in relation to the
			 * perimeter, the radius, r, is equal to p/(2*pi) and the area, a, is equal to p^2/(4*pi). 
			 * The number of calories in a pie is a * 10. Perry ate c out of t slices from the pie. 
			 * Thus the total number of calories Perry consumed is equal to (10*p^2*c)/(4*t*pi). 
			 * Round it up with Math.ceil or whatever way you prefer and print it out.
			 */
			double p = In.nextDouble(), t = In.nextDouble(), c = In.nextDouble();
			int caloriesConsumed = (int) Math.ceil(10*p*p/pi/4*c/t);
			System.out.println("Perfectly Popular Pizza "+currOrder+
					": Perry consumed "+caloriesConsumed+" calories.");

		}


	}

}